frame/recovery/src/lib.rs

```diff
@@ -245,8 +247,13 @@ pub mod pallet {
 		type FriendDepositFactor: Get<BalanceOf<Self>>;
 
 		/// The maximum amount of friends allowed in a recovery configuration.
+		///
+		/// NOTE: The threshold programmed in this Pallet uses u16, so it does
+		/// not really make sense to have a limit here greater than u16::MAX.
+		/// But also, that is a lot more than you should probably set this value
+		/// to anyway...
 		#[pallet::constant]
-		type MaxFriends: Get<u16>;
+		type MaxFriends: Get<u32>;
```

Comment:
Why did you change this to `u32`?

Reply:
BoundedVec needs a `Get<u32>`.

I tried to create a helper struct which maps a `Get<u16>` to a `Get<u32>`, but then a lot of compiler hell broke loose.

Please try it out if you think you know a good solution, otherwise, this should be fine.

Reply:
I also think this is fine in all practical purposes, it just makes me a bit uneasy, since the type was set to be `u16` for a reason, and I'd just wanna make sure that none of the original considerations were overlooked when changing it now to `u32`.

Reply:
> I tried to create a helper struct which maps a Get<u16> to a Get<u32>, but then a lot of compiler hell broke loose.

Yep just tried it as well. Apparently the `#[pallet::constant]`  looks out specifically for a `Get` identifier.